### PR TITLE
Change pipelines metric name regex limitation for kubeflow/pipelines#3741

### PR DESCRIPTION
**Pull request by Bobgy**
Upstream PR: kubeflow/pipelines#3741
Once it gets merged, we can merge this.
UPDATE: upstream PR merged.

### CODE REVIEW

File: content/en/docs/pipelines/sdk/pipelines-metrics.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ See the
 The metrics file has the following requirements:
 
 * The file path must be `/mlpipeline-metrics.json`.
-* `name` must follow the pattern `^[a-z]([-a-z0-9]{0,62}[a-z0-9])?$`.
+* `name` must follow the pattern `^[a-zA-Z]([-_a-zA-Z0-9]{0,62}[a-zA-Z0-9])?$`. (Until 0.5.2, `^[a-z]([-a-z0-9]{0,62}[a-z0-9])?$` must be followed).
```
Comment (joeliedtke) [inline content/en/docs/pipelines/sdk/pipelines-metrics.md:43]: Suggested revision:

* `name` must match the following pattern.
  `^[a-zA-Z]([-_a-zA-Z0-9]{0,62}[a-zA-Z0-9])?$`
  
  For Kubeflow Pipelines versions before 0.5.2, `name` must match the following pattern.
  `^[a-z]([-a-z0-9]{0,62}[a-z0-9])?$`

Comment (Bobgy) [reply content/en/docs/pipelines/sdk/pipelines-metrics.md:43]: Thanks! will udpate

**Comment by Bobgy** [comment]
@joeliedtke I updated, can you approve?
Thanks!

**Comment by joeliedtke** [comment]
Preview: https://deploy-preview-1950--competent-brattain-de2d6d.netlify.app/docs/pipelines/sdk/pipelines-metrics/

**Comment by joeliedtke** [comment]
/lgtm
/approve

Sorry for the delay, thanks @Bobgy!

**Comment by Bobgy** [comment]
/unhold
forgot about this for a while

**Pull request merged**